third_party/2and3/werkzeug/test.pyi

```diff
@@ -67,28 +70,62 @@ class EnvironBuilder:
+    response_wrapper: Type[_ResponseWrapper] = ...
     cookie_jar = ...  # type: Any
     allow_subdomain_redirects = ...  # type: Any
-    def __init__(self, application, response_wrapper: Optional[Any] = ..., use_cookies: bool = ...,
+    def __init__(self, application, response_wrapper: Optional[Type[_ResponseWrapper]] = ..., use_cookies: bool = ...,
                  allow_subdomain_redirects: bool = ...): ...
     def set_cookie(self, server_name, key, value: str = ..., max_age: Optional[Any] = ..., expires: Optional[Any] = ...,
                    path: str = ..., domain: Optional[Any] = ..., secure: Optional[Any] = ..., httponly: bool = ...,
                    charset: Text = ...): ...
     def delete_cookie(self, server_name, key, path: str = ..., domain: Optional[Any] = ...): ...
     def run_wsgi_app(self, environ, buffered: bool = ...): ...
     def resolve_redirect(self, response, new_location, environ, buffered: bool = ...): ...
-    def open(self, *args, **kwargs): ...
-    def get(self, *args, **kw): ...
-    def patch(self, *args, **kw): ...
-    def post(self, *args, **kw): ...
-    def head(self, *args, **kw): ...
-    def put(self, *args, **kw): ...
-    def delete(self, *args, **kw): ...
-    def options(self, *args, **kw): ...
-    def trace(self, *args, **kw): ...
+
+    @overload
+    def open(self, *args, as_tuple: Literal[True], **kwargs) -> Tuple[Mapping[str, Any], _ResponseWrapper]: ...
+    def open(self, *args, as_tuple: Literal[False] = ..., **kwargs) -> _ResponseWrapper: ...
+
+    @overload
+    def get(self, *args, as_tuple: Literal[True], **kw) -> Tuple[Mapping[str, Any], _ResponseWrapper]: ...
+    @overload
+    def get(self, *args, as_tuple: Literal[False] = ..., **kw) -> _ResponseWrapper: ...
+
+    def patch(self, *args, **kw) -> _ResponseWrapper: ...
+
+    @overload
+    def post(self, *args, as_tuple: Literal[True], **kw) -> Tuple[Mapping[str, Any], _ResponseWrapper]: ...
+    @overload
+    def post(self, *args, as_tuple: Literal[False] = ..., **kw) -> _ResponseWrapper: ...
+
+    @overload
+    def head(self, *args, as_tuple: Literal[True], **kw) -> Tuple[Mapping[str, Any], _ResponseWrapper]: ...
+    @overload
+    def head(self, *args, as_tuple: Literal[False] = ..., **kw) -> _ResponseWrapper: ...
+
+    @overload
+    def put(self, *args, as_tuple: Literal[True], **kw) -> Tuple[Mapping[str, Any], _ResponseWrapper]: ...
+    @overload
+    def put(self, *args, as_tuple: Literal[False] = ..., **kw) -> _ResponseWrapper: ...
+
+    @overload
+    def delete(self, *args, as_tuple: Literal[True], **kw) -> Tuple[Mapping[str, Any], _ResponseWrapper]: ...
+    @overload
+    def delete(self, *args, as_tuple: Literal[False] = ..., **kw) -> _ResponseWrapper: ...
+
+    @overload
+    def options(self, *args, as_tuple: Literal[True], **kw) -> Tuple[Mapping[str, Any], _ResponseWrapper]: ...
+    @overload
+    def options(self, *args, as_tuple: Literal[False] = ..., **kw) -> _ResponseWrapper: ...
```

Comment:
Literal overloads should always have a fallback with just `bool`, in case the type checker cannot infer the exact value.

Also, I'm not actually sure we allow `Literal` in typeshed yet.

Reply:
We should wait with new features, until our most important users (myoy, pytype, possibly pyre, PyCharm) support them, otherwise our stubs become unusable for them.

Reply:
Alright, I've dropped it for now and kept a TODO to fix it later. For now I'm going to assume that users of this library would only use the `as_tuple` argument to set it to True. The alternative is to force all callers to type the response.

Reply:
We actually allow Literal now @pcorpet.